src/components/profiles-modals/CreateProfileModal.vue

```diff
@@ -0,0 +1,89 @@
+<script lang="ts">
+import { Vue, Component } from 'vue-property-decorator';
+import { ModalCard } from "../all";
+import R2Error from "../../model/errors/R2Error";
+import sanitize from "sanitize-filename";
+
+@Component({
+    components: {ModalCard}
+})
+export default class CreateProfileModal extends Vue {
+    private newProfileName = '';
+    get isOpen(): boolean {
+        return this.$store.state.modals.isCreateProfileModalOpen;
+    }
+    closeModal() {
+        this.newProfileName = '';
+        this.$store.commit('closeCreateProfileModal');
+    }
+
+    get profileList(): string[] {
+        return this.$store.state.profiles.profileList;
+    }
+
+    makeProfileNameSafe(nameToSanitize: string): string {
+        return sanitize(nameToSanitize);
+    }
+
+    doesProfileExist(nameToCheck: string): boolean {
+        if ((nameToCheck.match(new RegExp('^([a-zA-Z0-9])(\\s|[a-zA-Z0-9]|_|-|[.])*$'))) === null) {
+            return true;
+        }
+        const safe: string = this.makeProfileNameSafe(nameToCheck);
+        return (this.profileList.some(function (profile: string) {
+            return profile.toLowerCase() === safe.toLowerCase()
+        }));
+    }
+
+    // User confirmed creation of a new profile with a name that didn't exist before.
+    // The profile can be either empty or populated via importing.
+    async createProfile() {
+        const safeName = this.makeProfileNameSafe(this.newProfileName);
+        if (safeName !== '') {
+            try {
+            document.dispatchEvent(new CustomEvent("created-profile", {detail: safeName}));
```

Comment:
Firing an event caught elsewhere is a bit spaghetty and not ideal from readability and maintenance perspectives. Ideally everything would be handled by the Vuex method.

Based on a quick peek, it looks the event listener might not be related to creating a blank profile at all, but for importing one. As the idea was to separate the modals from having shared code to make things clearer and more maintainable, having this modal to fire the event might not make sense.

So if the event is not required for this modal, remove the dispatch. If this modal actually needs something done in the event listener, see if it can be moved to be a part of the Vuex method instead. (Unless there's a good reason to do otherwise.)